File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/domain/SetEnableCallNotificationChannelDialogShownUseCase.java
package com.glia.widgets.core.dialog.domain;

import com.glia.widgets.core.dialog.PermissionDialogManager;

public class SetEnableCallNotificationChannelDialogShownUseCase {
    public final PermissionDialogManager permissionDialogManager;

    public SetEnableCallNotificationChannelDialogShownUseCase(PermissionDialogManager permissionDialogManager) {
        this.permissionDialogManager = permissionDialogManager;
    }

    public void execute() {
        this.permissionDialogManager.setEnableCallNotificationChannelRequestShown();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/core/dialog/domain/SetEnableCallNotificationChannelDialogShownUseCase.kt
package com.glia.widgets.core.dialog.domain

import com.glia.widgets.core.dialog.PermissionDialogManager

internal interface SetEnableCallNotificationChannelDialogShownUseCase {
    operator fun invoke()
}

internal class SetEnableCallNotificationChannelDialogShownUseCaseImpl(private val permissionDialogManager: PermissionDialogManager) :
    SetEnableCallNotificationChannelDialogShownUseCase {
    override fun invoke() = permissionDialogManager.setEnableCallNotificationChannelRequestShown()
}
